ui_framework/services/sort/sort_properties.test.js

```diff
@@ -0,0 +1,137 @@
+      const sortProperties = new SortProperties([name, size, color], 'size');
+      expect(sortProperties.getSortProperty().name).toBe('size');
+    });
+
+    test('throws an error with an invalid property name', () => {
+      expect(new SortProperties([name, size, color], 'does not exist')).toThrow();
+    });
+
+    test('is optional', () => {
+      const sortProperties = new SortProperties([name, size, color]);
+      expect(sortProperties.getSortProperty()).toBeUndefined();
+    });
+  });
+
+  describe('isAscendingByName', () => {
+    test('returns initial ascending values', () => {
+      const initialColorAscending = color.isAscending;
+      const initialNameAscending = name.isAscending;
+      const initialSizeAscending = size.isAscending;
+
+      const sortProperties = new SortProperties([name, size, color], 'color');
+
+      expect(sortProperties.isAscendingByName('color')).toBe(initialColorAscending);
+      expect(sortProperties.isAscendingByName('name')).toBe(initialNameAscending);
+      expect(sortProperties.isAscendingByName('size')).toBe(initialSizeAscending);
+    });
+  });
+
+  describe('sortOn', () => {
+    test('a new property name retains the original sort order', () => {
+      const initialColorAscending = color.isAscending;
+      const sortProperties = new SortProperties([name, size, color], 'color');
+      sortProperties.sortOn('name');
+      expect(sortProperties.isAscendingByName('color')).toBe(initialColorAscending);
+    });
+
+    test('the same property name twice flips sort order', () => {
+      const initialColorAscending = color.isAscending;
+      const sortProperties = new SortProperties([name, size, color], 'color');
+      sortProperties.sortOn('color');
+      expect(sortProperties.isAscendingByName('color')).toBe(!initialColorAscending);
+    });
+  });
+
+  describe('sortItems', () => {
+    test('sorts by initialSortProperty', () => {
+      const sortProperties = new SortProperties([name, size, color], 'name');
+      const sortedItems = sortProperties.sortItems(birds);
+      expect(sortedItems.length).toBe(birds.length);
+      expect(sortedItems[0].name).toBe('bluejay');
+      expect(sortedItems[1].name).toBe('cardinal');
+      expect(sortedItems[2].name).toBe('chickadee');
+    });
+
+    test('first sorts by initial isAscending value', () => {
+      const sortProperties = new SortProperties([name, size, color], 'size');
+      const sortedItems = sortProperties.sortItems(birds);
+      expect(sortedItems[0].size).toBe(8);
+      expect(sortedItems[1].size).toBe(7);
+      expect(sortedItems[2].size).toBe(3);
+    });
+
+    test('sorts by descending', () => {
+      const sortProperties = new SortProperties([name, size, color], 'name');
+      sortProperties.sortOn('size');
+      let sortedItems = sortProperties.sortItems(birds);
+      expect(sortedItems[0].size).toBe(8);
+      expect(sortedItems[1].size).toBe(7);
+      expect(sortedItems[2].size).toBe(3);
```

Comment:
I don't think we need these first three assertions since they're already covered by the preceding test.